lib/util/apply-disable-directives.js

```diff
@@ -104,11 +121,17 @@ module.exports = options => {
         }
 
         if (
-            globalDisableActive && enabledRules.has(problem.ruleId) ||
-            !globalDisableActive && !disabledRules.has(problem.ruleId)
+            enabledRules.has(problem.ruleId) ||
+            (!tempDisabledByLine.has("all") && !tempDisabledByLine.has(problem.ruleId) &&
+            !globalDisableActive && !disabledRules.has(problem.ruleId))
         ) {
             problems.push(problem);
         }
+
+        lineDisabledRules.set(problem.line, {
+            column: directive.column || 0,
```

Comment:
Why does the column matter here? `eslint-disable-line` comments should apply to the entire line, regardless of where they're placed on the line.